base_dav/views/dav_collection.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="view_dav_collection_tree" model="ir.ui.view">
+        <field name="model">dav.collection</field>
+        <field name="arch" type="xml">
+            <tree>
+                <field name="name"/>
+                <field name="dav_type"/>
+                <field name="model_id"/>
+                <field name="domain"/>
+            </tree>
+        </field>
+    </record>
+    <record id="view_dav_collection_form" model="ir.ui.view">
+        <field name="model">dav.collection</field>
+        <field name="arch" type="xml">
+            <form>
+                <group>
+                    <field name="name"/>
+                    <field name="dav_type"/>
+                    <field name="url" widget="url" attrs="{'invisible': [('url', '=', False)]}"/>
+                </group>
+                <group string="Access">
+                    <field name="model_id"/>
+                    <!-- TODO: use widget="domain" when we can set the model dynamically /-->
+                    <field name="domain"/>
+                    <field name="field_uuid" domain="[('model_id', '=', model_id)]"/>
+                    <field name="rights"/>
+                </group>
+                <group string="Additional field mapping">
+                    <field name="field_mapping_ids" nolabel="1"/>
+                </group>
+            </form>
+        </field>
+    </record>
+
+    <record id="view_dav_collection_mapping_tree" model="ir.ui.view">
+        <field name="model">dav.collection.field_mapping</field>
+        <field name="arch" type="xml">
+            <tree>
+                <field name="name"/>
+                <field name="field_id"/>
+                <field name="mapping_type"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_dav_collection_mapping_form" model="ir.ui.view">
+        <field name="model">dav.collection.field_mapping</field>
+        <field name="arch" type="xml">
+            <form>
+                <group>
+                    <field name="collection_id" invisible="1"/>
```

Comment:
When trying to create a field mapping from the view, we got

```
The following fields are invalid:

Collection
```

Default value must be set on parent view.

Or if that view is only used embedded in the collection view you can try to remove this field completely. Not sure it would work with a Form view but Odoo is sets automatically the field like on `account.invoice.line` without the default but it's an inline edition with a tree view. There are some magic behind that and I don't know on which conditions it might not work with Form view, thus probably safer to set the default in context as in `res.partner`.